app/models/motd_file.rb

```diff
@@ -1,67 +1,43 @@
+require 'open-uri'
+
 class MotdFile
-  attr_reader :motd_system_file
 
-  Message = Struct.new :date, :title, :body do
-    def self.from(str)
-      if str =~ /(\d+[-\/\.]\d+[-\/\.]\d+)\n--- ([ \S ]*)\n(.*)/m
-        MotdFile::Message.new(Date.parse($1), $2, $3)
-      else
-        nil
-      end
-    rescue ArgumentError => e
-      Rails.logger.warn("MOTD message poorly formatted: #{e} => #{e.message}")
+  attr_reader :motd_uri
 
-      nil
-    end
-  end
   # Initialize the Motd Controller object based on the current user.
   #
-  # @param [boolean] update_user_view_timestamp True to update the last viewed timestamp. (Default: false)
-  def initialize(path = ENV['MOTD_PATH'], update_user_view_timestamp: false)
-    @motd_system_file = path
-
-    touch if update_user_view_timestamp
-  end
-
-  # An empty file whose modification timestamp indicates the last time the user
-  # viewed the motd messages. This is useful for when we want to use the file
-  # system to determine when new messages the user has not seen have been
-  # added to the motd.
-  def motd_config_file
-    @motd_config_file ||= OodAppkit.dataroot.join(".motd")
+  # @param [String] path The path to the motd file as a URI
+  def initialize(path = ENV['MOTD_PATH'])
+    @motd_uri = path
   end
 
+  # Checks the path URI to see if it can be opened
+  #
+  # Uses open-uri to check local or remote path for contents
   def exist?
-    motd_system_file && File.file?(motd_system_file)
+    exists = false
+    if motd_uri
+      begin
+        open(motd_uri)
+        exists = true
+      rescue
+        # The messages file does not exist.
+        Rails.logger.warn "MOTD File is missing; it was expected at #{motd_uri}"
+        exists = false
+      end
+    end
+    exists
   end
 
-  # If the motd file hasn't been created on the system, or if the system motd is newer than the user's file, return true.
-  def new_messages?
-    # FIXME: Use if/else statements because this is arcane
-    (messages.count > 0) ? ( !File.exist?(motd_config_file) ? true : File.new(motd_system_file).ctime > File.new(motd_config_file).ctime ? true : false ) : false   
+  # Create an array of message objects based on the current message of the day.
+  def content
+    if self.exist?
```

Comment:
The first link doesn't exist in this PR anymore:

https://github.com/OSC/ood-dashboard/blob/e68a8e334b60cafd817ec118a6a5cb9336d0fd7d/app/views/dashboard/index.html.erb#L7

and the second link will actually be a bug for this PR.